mplc/splitter.py

```diff
@@ -89,6 +82,69 @@ def copy(self):
         return self.__copy__()
 
 
+class FlexibleSplitter(Splitter):
+    name = 'Fully flexible samples split'
+
+    def __init__(self, amounts_per_partner, configuration, **kwargs):
+
+        logger.info("Proceeding to a flexible split as requested. Please note that the flexible "
+                    "split currently discards the amounts_per_partner (if provided) and infers amounts of samples "
+                    "per partner from the samples_split_configuration provided.")
+
+        # First we re-assemble the split configuration per cluster, and check its coherence
+        self.samples_split_grouped_by_cluster = list(zip(*configuration))
+        for idx, cluster in enumerate(self.samples_split_grouped_by_cluster):
+            if np.sum(cluster) > 1:
+                raise ValueError(f"Amounts of samples of class {idx} split among partners exceed 100%, "
+                                 f"the dataset split cannot be performed.")
+
```

Comment:
You could probably vectorize this, as the samples_split_grouped_by_cluster is a list of list (ie np.array)

Reply:
Yes - unless it really provides an improvement, I'll stick to the `list(zip(*...))` for now